activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb

```diff
@@ -56,11 +56,16 @@ def concise_options(options)
     # +columns+ attribute of said TableDefinition object, in order to be used
     # for generating a number of table creation or table changing SQL statements.
     ColumnDefinition = Struct.new(:name, :type, :options, :sql_type) do # :nodoc:
+      self::OPTIONS = %i(limit precision scale default null collation comment
```

Comment:
Hadn't seen this `self::` construct before but it's apparently required for `Structs`:
https://github.com/rails/rails/commit/3d91649654ff5570d0cf26f96ec9f82f31fe32f6

Reply:
Yea it looks like it has something to do with constant resolution and how Structs dont create a namespace like modules and classes do. https://stackoverflow.com/a/53727917